bluezero/adapter.py

```diff
@@ -258,10 +259,16 @@ def _properties_changed(self, interface, changed, invalidated, path):
         Handle DBus PropertiesChanged signal and
         call appropriate user callback
         """
-        if self.on_disconnect is not None:
-            if 'Connected' in changed:
-                if not changed['Connected']:
-                    self.on_disconnect()
+        macaddr = dbus_tools.get_mac_addr_from_dbus_path(path)
+        if 'Connected' in changed:
+            new_dev = device.Device(
+                adapter_addr=self.address,
+                device_addr=macaddr)
+            self.on_device_found(new_dev)
```

Comment:
We don't need the `self.on_device_found(new_dev)` line here.

Reply:
damned! I copied from the bottom , sorry : fixed